pinot-common/src/main/java/org/apache/pinot/common/lineage/SegmentLineage.java

```diff
@@ -146,4 +150,19 @@ public ZNRecord toZNRecord() {
     }
     return znRecord;
   }
+
+  /**
+   * Returns a json representation of the segment lineage.
+   * Segment lineage entries are sorted in chronological order by default.
+   */
+  public ObjectNode toJsonObject() {
+    ObjectNode jsonObject = JsonUtils.newObjectNode();
+    jsonObject.put("tableNameWithType", _tableNameWithType);
+    LinkedHashMap<String, LineageEntry> sortedLineageEntries = new LinkedHashMap<>();
+    _lineageEntries.entrySet().stream()
+        .sorted(Map.Entry.comparingByValue(Comparator.comparingLong(LineageEntry::getTimestamp)))
+        .forEachOrdered(x -> sortedLineageEntries.put(x.getKey(), x.getValue()));
```

Comment:
@snleee fyi ensures resulting entries of listing segment lineage REST API are in chronological order here.

Reply:
IMO sorting is not necessary. at least we shouldn't do this in the toJsonObject path. what do you think we add the timestamp processing when we support filtering arguments, for example: `segments/{tableName}/lineage?timestamp>-10days`

Reply:
discussed offline. We agreed to file the issue and address it from the separate PR.